x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EsQueryExec.java

```diff
@@ -332,4 +334,8 @@ public String searchOrder() {
             return searchOrder;
         }
     }
+
+    public boolean scoring() {
```

Comment:
IMO scoring isn't really a property of this class, but that of an attribute within. I'd do this check where needed (one place so far) and remove the method (ideally deduplicating the looping code in `ReplaceSourceAttributes` as well, but only if not too cumbersome).